JoinRpg.Domain/SubscribeExtensions.cs

```diff
@@ -9,40 +9,69 @@ namespace JoinRpg.Domain
   {
-    public static IEnumerable<User> GetSubscriptions(this ForumThread claim, [CanBeNull] IEnumerable<User> extraRecepients, bool isVisibleToPlayer)
+    public static IEnumerable<User> GetSubscriptions(
+      this ForumThread forumThread,
+      [CanBeNull] IEnumerable<User> extraRecipients,
+      bool isVisibleToPlayer)
     {
       return
-        claim.Subscriptions //get subscriptions on forum
+        forumThread.Subscriptions //get subscriptions on forum
           .Select(u => u.User) //Select users
-          .Union(extraRecepients ?? Enumerable.Empty<User>()) //add extra recepients
-          .VerifySubscriptions(isVisibleToPlayer, claim);
+          .Union(extraRecipients ?? Enumerable.Empty<User>()) //add extra recipients
+          .VerifySubscriptions(isVisibleToPlayer, forumThread);
+    }
+
+    public static IEnumerable<User> GetSubscriptions(
+      this Character character,
+      Func<UserSubscription, bool> predicate,
+      [CanBeNull] IEnumerable<User> extraRecipients = null,
+      bool mastersOnly = false
+      )
+    {
+      if (character == null) return Enumerable.Empty<User>();
+      return character.GetGroupsPartOf() //Get all groups for the character
+        .SelectMany(g => g.Subscriptions) //get subscriptions on groups
+        .Union(character.Subscriptions) //Subscriptions of the character itself.
+        .Union(character.ApprovedClaim?.Subscriptions ?? Enumerable.Empty<UserSubscription>()) //Subscriptions of the claim itself.
+        .Where(predicate) //type of subscribe (on new comments, on new claims etc.)
+        .Select(u => u.User) //Select users
+        .Union(character.ApprovedClaim?.Player) //...and player who claimed for the character
+        .Union(character.ApprovedClaim?.ResponsibleMasterUser) //claim esponsible master is always subscribed on everything related to the claim
+        .Union((character.ResponsibleMasterUser)) //...and the measter who's responsible for the character
+        .Union(extraRecipients ?? Enumerable.Empty<User>()) //add extra recipients
+        .VerifySubscriptions(mastersOnly, character)
+        .Distinct(); //we make union of subscriptions and directly taken users. Duplicates may appear.
     }
 
     public static IEnumerable<User> GetSubscriptions(
       this Claim claim, 
       Func<UserSubscription, bool> predicate,
-      [CanBeNull] IEnumerable<User> extraRecepients, 
-      bool isVisibleToPlayer)
+      [CanBeNull] IEnumerable<User> extraRecipients = null, 
+      bool mastersOnly = false)
     {
       return claim.GetTarget().GetGroupsPartOf() //Get all groups for claim
           .SelectMany(g => g.Subscriptions) //get subscriptions on groups
           .Union(claim.Subscriptions) //subscribtions on claim
-          .Union(claim.Character?.Subscriptions ?? new UserSubscription[] {}) //and on characters
+          .Union(claim.Character?.Subscriptions ?? Enumerable.Empty<UserSubscription>()) //and on characters
```

Comment:
added but kept copy-paste in place because it's nto a pure copypaste. I've left a TODO to think later - now I'm trying to complete my changes before you invented smth new to merge with :-)